examples/Reranking_search_results_with_cross-encoders.ipynb

```diff
@@ -0,0 +1,805 @@
+      "12: Learning Joint Representations of Videos and Sentences with Web Image Search\n",
+      "13: Character-based Neural Networks for Sentence Pair Modeling\n",
+      "14: Train Once, Test Anywhere: Zero-Shot Learning for Text Classification\n",
+      "15: Hierarchical GPT with Congruent Transformers for Multi-Sentence Language Models\n",
+      "16: Sentence-T5: Scalable Sentence Encoders from Pre-trained Text-to-Text Models\n",
+      "17: In Search for Linear Relations in Sentence Embedding Spaces\n",
+      "18: Learning to Borrow -- Relation Representation for Without-Mention Entity-Pairs for Knowledge Graph Completion\n",
+      "19: Efficient and Flexible Topic Modeling using Pretrained Embeddings and Bag of Sentences\n",
+      "20: Relational Sentence Embedding for Flexible Semantic Matching\n"
+     ]
+    }
+   ],
+   "source": [
+    "for i, result in enumerate(result_list):\n",
+    "    print(f\"{i + 1}: {result['title']}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d5727678",
+   "metadata": {},
+   "source": [
+    "## Cross-encoder\n",
+    "\n",
+    "We'll create a cross-encoder using the ```Completions``` endpoint - the key factors to consider here are:\n",
+    "- Make your examples domain-specific - the strength of cross-encoders comes when you fine-tune them to your domain.\n",
+    "- There is a trade-off between how many potential examples to re-rank vs. processing speed. Consider batching and parallel processing cross-encoder requests to aid user experience.\n",
+    "\n",
+    "The steps here are:\n",
+    "- Build a prompt to assess relevance and provide few-shot examples to tune it to your domain\n",
+    "- Return the classification of yes/no as well as the ```logprobs```\n",
+    "- Rerank the results by the ```logprobs``` keyed on Yes\n",
+    "- Add a ```logit bias``` for the tokens for Yes and No to decrease the likelihood of any other tokens occurring"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "ca634bf9",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "([5297], [2949])"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "tokens = ['Yes','No']\n",
+    "tokenizer = tiktoken.get_encoding(\"gpt2\")\n",
+    "ids = [tokenizer.encode(token) for token in tokens]\n",
+    "ids[0],ids[1]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "4fdf8c11",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "prompt = '''\n",
+    "You are an Assistant responsible for helping detect whether the retrieved document is relevant to the query. For a given input, you need to output a single token: \"Yes\" or \"No\" indicating the retrieved document is relevant to the query.\n",
+    "\n",
+    "Examples:\n",
```

Comment:
I highly, highly recommend not telling the model that these are examples and that the final query is different. Best practice is to format the final query exactly as the examples to minimize the odds of treating it differently.

So concretely, I would:
- Remove the Examples label
- Remove the End of Examples label
- Remove the Begin label
- And just make the final query look like a half-written example